src/foam/nanos/notification/email/EmailTemplate.js

```diff
@@ -54,16 +82,160 @@ bodyAsByteArrayIsSet_ = false;`
       transient: true,
       type: 'Byte[]',
       javaFactory: 'return getBody() != null ? getBody().getBytes(StandardCharsets.UTF_8) : null;'
-    },
+    }
+  ],
+  methods: [
     {
-      class: 'String',
-      name: 'displayName'
+      name: 'apply',
+      documentation: `Throws exception if any errors - calling Service will/should catch.`,
+      type: 'foam.nanos.notification.email.EmailMessage',
+      javaThrows: ['java.lang.NoSuchFieldException'],
+      args: [
+        {
+          name: 'x',
+          type: 'Context',
+        },
+        {
+          name: 'user',
+          type: 'foam.nanos.auth.User',
+          documentation: 'user whose the recipient of the email being sent'
+        },
+        {
+          name: 'emailMessage',
+          type: 'foam.nanos.notification.email.EmailMessage',
+          documentation: 'Email message'
+        },
+        {
+          name: 'templateArgs',
+          type: 'Map',
+          documentation: 'Template arguments'
+        },
+        {
+          name: 'config',
+          javaType: 'org.jtwig.environment.EnvironmentConfiguration'
+        }
+      ],
+      javaCode: `
+        // Check basic properties
+        if ( emailMessage == null ) throw new NoSuchFieldException("emailMessage is Null");
+
+        String tempKeyString = "";
+        Object value = null;
+
+        // process templateArgs
+        for ( Object key : templateArgs.keySet() ) {
+          value = templateArgs.get((String)key);
+          if ( value instanceof String ) {
+            tempKeyString = (String) value;
+            templateArgs.put((String) key, new String(tempKeyString.getBytes(StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8));
+          }
+        }
+
+        // build model, and confirm it was built
+        JtwigModel model = JtwigModel.newModel(templateArgs);
+        if ( model == null ) throw new NoSuchFieldException("JtwigModel is Null");
+
+        // Try to locate user group
+        DAO groupDAO = ((DAO) x.get("groupDAO")).inX(x);
+        Group group = null;
```

Comment:
You can do:
```
Group group = user != null ? (Group) groupDAO.find(user.getGroup()) : null
```
And then remove the next 3 lines.

Though there's quite a bit of logic going on here for simply finding the group and I wonder if it's all necessary. Don't we usually put a group in the context and could we just be using that? If it's not in the context, is that a sign that something's wrong or is falling back to the user's group reasonable? Also, when would `groupDAO.find(user.getGroup())` ever return null?